Gulpfile.js

```diff
@@ -70,3 +77,96 @@ gulp.task("watch", ["build"], function() {
     gulp.start("build");
   });
 });
+
+gulp.task("build-babel-standalone", cb => {
+  pump(
+    [
+      gulp.src(__dirname + "/packages/babel-standalone/src/index.js"),
+      webpackBuild(),
+      gulp.dest(__dirname + "/packages/babel-standalone"),
+      uglify(),
+      rename({ extname: ".min.js" }),
+      gulp.dest(__dirname + "/packages/babel-standalone"),
+    ],
+    cb
+  );
+});
+
+function webpackBuild() {
+  let version = require("./packages/babel-core/package.json").version;
+
+  // If this build is part of a pull request, include the pull request number in
+  // the version number.
+  if (process.env.CIRCLE_PR_NUMBER) {
+    version += "+pr." + process.env.CIRCLE_PR_NUMBER;
+  }
+
+  const config = {
+    module: {
+      rules: [
+        {
+          test: /\.js$/,
+          include: /node_modules/,
+          loader: "babel-loader",
+          options: {
+            // Some of the node_modules may have their own "babel" section in
+            // their project.json (or a ".babelrc" file). We need to ignore
+            // those as we're using our own Babel options.
+            babelrc: false,
+            presets: ["env"],
+          },
+        },
+        {
+          test: /\.js$/,
+          exclude: /node_modules/,
+          loader: "babel-loader",
+          options: {
+            // Some of the node_modules may have their own "babel" section in
+            // their project.json (or a ".babelrc" file). We need to ignore
+            // those as we're using our own Babel options.
+            babelrc: false,
+            presets: ["env", "stage-0"],
+          },
+        },
+      ],
+    },
+    node: {
+      // Mock Node.js modules that Babel require()s but that we don't
+      // particularly care about.
+      fs: "empty",
+      module: "empty",
+      net: "empty",
+    },
+    output: {
+      filename: "babel.js",
+      library: "Babel",
+      libraryTarget: "umd",
+    },
+    plugins: [
+      new webpack.DefinePlugin({
+        "process.env.NODE_ENV": '"production"',
+        BABEL_VERSION: JSON.stringify(version),
+        VERSION: JSON.stringify(version),
+      }),
+      /*new webpack.NormalModuleReplacementPlugin(
+        /..\/..\/package/,
+        "../../../../src/babel-package-shim"
+      ),*/
+      new webpack.optimize.ModuleConcatenationPlugin(),
+    ],
+    resolve: {
+      plugins: [
+        // Dedupe packages that are used across multiple plugins.
+        // This replaces DedupePlugin from Webpack 1.x
+        new RootMostResolvePlugin(__dirname, true),
+      ],
+    },
+  };
+
+  return webpackStream(config, webpack);
+  // To write JSON for debugging:
+  /*return webpackStream(config, webpack, (err, stats) => {
+    require('gulp-util').log(stats.toString({colors: true}));
+    require('fs').writeFileSync('webpack-debug.json', JSON.stringify(stats.toJson()));
+  });*/
+}
```

Comment:
Is it possible to still keep this in a file that can be exported, so that other projects can use it? For example, see https://github.com/babel/minify/pull/679

Reply:
I don't really understand how this could be made reusable. All the packages are devDeps, so none of the things used in here will actually be installed if you `npm install babel-standalone`, and since it's not a static list, if someone did depend on it and we added a new package to `babel-standalone`, it'd be a breaking change for anyone depending on it.

I'm also not totally clear on the usecase of reusing this. Who are we targeting, and how do they know this even exists as a public API?

Reply:
The use case is currently just for Babel projects that aren't in the monorepo (mainly babel-minify and babel-preset-env). In the past, since babel-standalone was a separate repo, it could build packages for all three (babel-standalone, babel-minify-standalone, babel-preset-env-standalone) all in that repo. Now that babel-standalone is in this repo, we can't build babel-minify and babel-preset-env from here. However, those two projects still want to reuse the same Gulpfile + Webpack configuration, as everything is exactly the same, and it has all the configuration that's common across *all* babel-standalone plugins/presets.

There's likely a better way of sharing the config - I'm open to suggestions. The devDeps are an issue, you're right.